sapi/fpm/fpm/fpm_main.c

```diff
@@ -1918,6 +1918,8 @@ consult the installation file that came with this distribution, or visit \n\
 
 			php_execute_script(&file_handle);
 
+			zend_destroy_file_handle(&file_handle);
```

Comment:
Doing destroy when it is already in list is not necessary and it will be slower because it needs to remove the element from the list. I think it makes sense only if it is not in the list. I checked with opcache and it is set to false on subsequent runs so this should work fine.
```suggestion
			if (!file_handle.in_list) {
				zend_destroy_file_handle(&file_handle);
			}
```